diff --git a/files/run b/files/run
index 69f0705..b7edd50 100755
--- a/files/run
+++ b/files/run
@@ -3,6 +3,10 @@
 dir=/var/lib/rkt/pods/run
 uuid=${@: -1}
 pod=$dir/$uuid
+if [ ! -d "$pod" ]; then
+  dir=$(dirname "$(dirname "$(dirname "$(dirname "$(readlink -f "$0")")")")")
+  pod=$dir/$uuid
+fi
 stage1=$pod/stage1/rootfs
 # Assumes 1 App
 stage2m=$pod/pod
diff --git a/files/stop b/files/stop
index 5ca1cbe..0352cf5 100755
--- a/files/stop
+++ b/files/stop
@@ -1,4 +1,9 @@
 #!/bin/bash
-
-xl destroy $(cat "/var/lib/rkt/pods/run/$1/pid") &>/dev/null
+pidfile=/var/lib/rkt/pods/run/$1/pid
+if [[ -f $pidfile ]]; then
+  xl destroy $(cat "$pidfile") &>/dev/null
+else
+  pidfile="$(dirname "$0")/../../pid"
+  xl destroy $(cat "$pidfile") &>/dev/null
+fi
 exit 0
